### PR TITLE
[HttpClient] Fix seeking in not-yet-initialized requests

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Reported on Slack by @drupol

### CODE REVIEW

**Comment by drupol** [comment]
Oh ! Going to test this ASAP ! Thanks @nicolas-grekas !

**Review by stof — APPROVED** [review]

File: src/Symfony/Component/HttpClient/Response/StreamWrapper.php [hunk 1]
```diff
@@ -81,6 +81,7 @@ public function bindHandles(&$handle, &$content): void
     {
         $this->handle = &$handle;
         $this->content = &$content;
+        $this->offset = null;
```
Comment (drupol) [inline src/Symfony/Component/HttpClient/Response/StreamWrapper.php:84]: How about initializing it to zero directly?

**Comment by drupol** [comment]
I tested it and it works, however, there is still an issue in our app.

This is because of [PsrHttpFactory.php#L130](https://github.com/symfony/psr-http-message-bridge/blob/main/Factory/PsrHttpFactory.php#L130). I have the feeling that it also need some adaptations.

**Comment by drupol** [comment]
I made further test, but I'm still unable to get the body of a response twice. I'm also unable to reproduce the issue in a few lines of code.

I made a screenshot after adding some `dump` and `dd` in `Psr18Client::sendRequest()`, without changing anything else, as such:

```php
   /**
     * {@inheritdoc}
     */
    public function sendRequest(RequestInterface $request): ResponseInterface
    {
        try {
            $body = $request->getBody();

            if ($body->isSeekable()) {
                $body->seek(0);
            }

            $response = $this->client->request($request->getMethod(), (string) $request->getUri(), [
                'headers' => $request->getHeaders(),
                'body' => $body->getContents(),
                'http_version' => '1.0' === $request->getProtocolVersion() ? '1.0' : null,
            ]);

            $psrResponse = $this->responseFactory->createResponse($response->getStatusCode());

            foreach ($response->getHeaders(false) as $name => $values) {
                foreach ($values as $value) {
                    try {
                        $psrResponse = $psrResponse->withAddedHeader($name, $value);
                    } catch (\InvalidArgumentException $e) {
                        // ignore invalid header
                    }
                }
            }

            if ($response instanceof StreamableInterface) {
                dump('Response is StreamableInterface', $response);
                $body = $response->toStream(false);
            } else {
                dump('Response is not StreamableInterface', $response);
                $body = StreamWrapper::createResource($response, $this->client);
            }

            $body = $this->streamFactory->createStreamFromResource($body);

            $r = $psrResponse->withBody($body);

            dump((string)$r->getBody());
            dd((string)$r->getBody());

            return $r;
        } catch (TransportExceptionInterface $e) {
            if ($e instanceof \InvalidArgumentException) {
                throw new Psr18RequestException($e, $request);
            }

            throw new Psr18NetworkException($e, $request);
        }
    }
```
 
And here's the result. Sorry for the pixelisation of the information, but they are not relevant here.

screenshot-127 0 0 1_8000-2022 10 07-23_45_48

**Comment by nicolas-grekas** [comment]
Please provide a reproducing script if you can, I don't be able to
understand without one.

**Comment by drupol** [comment]
I've spent the whole evening trying to reproduce it, without luck. I'll try again this weekend.

**Comment by drupol** [comment]
What I propose is to merge this PR, then I'll open another issue for the rest. WDYT ?

**Review by chalasr — APPROVED** [review]

**Review by mtarld — APPROVED** [review]

**Pull request merged**